### PR TITLE
Feature/teku integration

### PR DESCRIPTION
**Pull request by Wolmin**
https://app.clickup.com/t/2645698/DEV-5909

DEV-5909

### CODE REVIEW

**Comment by Wolmin** [comment]
Note: this PR only includes support for Teku beacon client - we're still waiting for input on how do we want to manage validator keystores

File: commands/init.go [hunk 1]
```diff
@@ -61,6 +61,10 @@ func InitializeDirectory(ctx *cli.Context) error {
 	_ = initConfigGroup(configs.PrysmValidatorConfigDependencies)
 	log.Info("✅  Prysm validator configuration files downloaded!\n\n")
 
+	log.Info("⬇️  Downloading teku configuration files...")
+	_ = initConfigGroup(configs.TekuConfigDependencies)
+	log.Info("✅  Prysm validator configuration files downloaded!\n\n")
```
Comment (kalote) [inline commands/init.go]: should it be `Teku validator configuration files downloaded!` ?

File: dependencies/clients/teku.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+	"github.com/lukso-network/tools-lukso-cli/common/utils"
+	"github.com/lukso-network/tools-lukso-cli/flags"
+	"github.com/lukso-network/tools-lukso-cli/pid"
+)
+
+const (
+	tekuDepsFolder = "teku" // folder in which both teku and JDK are stored
+	tekuFolder     = "teku" // folder in which teku is stored (in tekuDepsFolder)
+	jdkFolder      = "jdk"  // folder in which JDK is stored (in tekuDepsFolder)
+	jdkInstallURL  = "https://download.java.net/java/GA/jdk20.0.2/6e380f22cbe7469fa75fb448bd903d8e/9/GPL/openjdk-20.0.2_linux-x64_bin.tar.gz"
```
Comment (kalote) [inline dependencies/clients/teku.go]: we should stick to the version used by Teku, which is 17.

File: dependencies/clients/teku.go [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+		log.Infof("✅  %s updated!\n\n", t.Name())
+	case false:
+		log.Infof("✅  %s downloaded!\n\n", t.Name())
+	}
+
+	_, isInstalled := os.LookupEnv(system.JavaHomeEnv) // means that JDk is not set up
+	if !isInstalled {
+		message := "Teku is written in Java. This means that to use it you need to have:\n" +
+			"- JDK installed on your computer\n" +
+			"- JAVA_HOME environmental variable set\n" +
```
Comment (kalote) [inline dependencies/clients/teku.go]: "- JAVA_HOME environment variable set\n" +

File: dependencies/clients/teku.go [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+	log.Info("✅  JDK downloaded!\n\n")
+
+	luksoNodeDir, err := os.Getwd()
+	if err != nil {
+		return
+	}
+
+	javaHomeVal := fmt.Sprintf("%s/%s/%s", luksoNodeDir, tekuDepsFolder, jdkFolder)
+
+	log.Infof("⚙️  To continue working with Teku please export the JAVA_HOME environmental variable.\n"+
```
Comment (kalote) [inline dependencies/clients/teku.go]: environmental => environment

File: dependencies/clients/teku.go [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+
+	if response.StatusCode == http.StatusNotFound {
+		log.Warnf("⚠️  File under URL %s not found - skipping...", url)
+
+		return nil
+	}
+
+	if http.StatusOK != response.StatusCode {
+		return fmt.Errorf(
+			"❌  Invalid response when downloading on file URL: %s. Response code: %s",
```
Comment (kalote) [inline dependencies/clients/teku.go]: Invalid response while downloading the file at URL:

File: dependencies/clients/teku.go (hunk 2 above)
Comment (Wolmin) [reply dependencies/clients/teku.go]: `Building on a more recent version of Java is supported, but the resulting build will not work on earlier versions of Java.`
Honestly I think that this 20 version is fine - jdk.java.net doesn't even support version 17 at the moment

File: dependencies/clients/teku.go (hunk 3 above)
Comment (Wolmin) [reply dependencies/clients/teku.go]: Done

File: dependencies/clients/teku.go (hunk 4 above)
Comment (Wolmin) [reply dependencies/clients/teku.go]: Done

File: dependencies/clients/teku.go (hunk 5 above)
Comment (Wolmin) [reply dependencies/clients/teku.go]: Done

File: commands/init.go (hunk 1 above)
Comment (Wolmin) [reply commands/init.go]: Yes, you are right, adjusted :+1:

**Review by mxmar — CHANGES_REQUESTED** [review]
@Wolmin Please review `README.md` for necessary `Teku and Teku Validator` changes.

File: dependencies/clients/teku.go [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+type TekuClient struct {
+	*clientBinary
+}
+
+func NewTekuClient() *TekuClient {
+	return &TekuClient{
+		&clientBinary{
+			name:           tekuDependencyName,
+			commandName:    "teku",
+			baseUrl:        "https://artifacts.consensys.net/public/teku/raw/names/teku.tar.gz/versions/|TAG|/teku-|TAG|.tar.gz",
```
Comment (mxmar) [inline dependencies/clients/teku.go]: Please extract to `dependencies/clients/teku.go` consts if possible

File: README.md [hunk 7]
```diff
@@ -13,8 +13,8 @@ The LUKSO CLI is a command line tool to install, manage and set up validators of
 The LUKSO CLI is able to install multiple clients for running the node.
 
 - Execution Clients: [Geth](https://geth.ethereum.org/), [Erigon](https://github.com/ledgerwatch/erigon)
-- Consensus Clients: [Prysm](https://github.com/prysmaticlabs/prysm), [Lighthouse](https://github.com/sigp/lighthouse)
-- Validator Staking Clients: [Prysm](https://docs.prylabs.network/docs/how-prysm-works/prysm-validator-client), [Lighthouse](https://github.com/sigp/lighthouse)
+- Consensus Clients: [Prysm](https://github.com/prysmaticlabs/prysm), [Lighthouse](https://github.com/sigp/lighthouse), [Teku](https://github.com/Consensys/teku)
```
Comment (mxmar) [inline README.md:16]: We need to explain how to prepare local machine for `Teku` in `README` (JAVA!)

File: README.md [hunk 8]
```diff
@@ -455,10 +455,10 @@ Because different clients have different processes of exiting please make sure t
 Make sure your validator node is running before starting the exit command.
 
 ```sh
-# Exit validator keys with Prysm for Mainnet
+# Exit validator keys with Prysm/Teku for Mainnet
```
Comment (mxmar) [inline README.md:463]: Please extend `validator exit` descriptions for all consensus clients (how it works and if it affects all keys, or you need to select - you can explain what will happen when you execute `sudo lukso validator exit` - about external consensus client CLI

File: dependencies/clients/teku.go [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+	case false:
+		log.Infof("✅  %s downloaded!\n\n", t.Name())
+	}
+
+	_, isInstalled := os.LookupEnv(system.JavaHomeEnv) // means that JDk is not set up
+	if !isInstalled {
+		message := "Teku is written in Java. This means that to use it you need to have:\n" +
+			"- JDK installed on your computer\n" +
+			"- JAVA_HOME environment variable set\n" +
+			"Do you want to install and set up JDK along with Teku? [Y/n]\n>"
```
Comment (mxmar) [inline dependencies/clients/teku.go:83]: We need to mention about JAVA in `README`

Comment (Wolmin) [reply dependencies/clients/teku.go:83]: Done

File: README.md (hunk 7 above)
Comment (Wolmin) [reply README.md:16]: User doesn't need to prepare anything - he will install the JDK when installing Teku, so this is covered - info about potential dependencies needed are present in the README.

File: dependencies/clients/teku.go (hunk 6 above)
Comment (Wolmin) [reply dependencies/clients/teku.go]: Done

File: README.md (hunk 8 above)
Comment (Wolmin) [reply README.md:463]: Done

**Review by mxmar — APPROVED** [review]

**Pull request merged**